From 5e91b7977570096c9a788b455ac65221f57599af Mon Sep 17 00:00:00 2001
From: Manuel Rego Casasnovas <mrego@igalia.com>
Date: Wed, 14 Oct 2009 08:00:56 +0200
Subject: [PATCH] ItEr30S11CUVistaRecursosTempoEmpresaItEr29S13: Checking if there are or not day assignments.

---
 .../web/planner/OrderPlanningModel.java            |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/navalplanner-webapp/src/main/java/org/navalplanner/web/planner/OrderPlanningModel.java b/navalplanner-webapp/src/main/java/org/navalplanner/web/planner/OrderPlanningModel.java
index 2e9b1f9..74225e0 100644
--- a/navalplanner-webapp/src/main/java/org/navalplanner/web/planner/OrderPlanningModel.java
+++ b/navalplanner-webapp/src/main/java/org/navalplanner/web/planner/OrderPlanningModel.java
@@ -443,6 +443,10 @@ public abstract class OrderPlanningModel implements IOrderPlanningModel {
     private String getServletUri(
             final SortedMap<LocalDate, Integer> mapDayAssignments,
             final Date start, final Date finish) {
+        if (mapDayAssignments.isEmpty()) {
+            return "";
+        }
+
         setMaximunValueForChartIfGreater(Collections.max(mapDayAssignments.values()));
 
         String uri = CallbackServlet
-- 
1.6.0.4

