From 57213fd1d6ef551b5e1301da2dc459e9e68c2508 Mon Sep 17 00:00:00 2001
From: Manuel Rego Casasnovas <mrego@igalia.com>
Date: Tue, 13 Oct 2009 09:11:49 +0200
Subject: [PATCH] ItEr30S17ValidacionEProbasFuncionais: Showing proper error information creating an order. Fixing Bug #5.

---
 .../web/orders/OrderCRUDController.java            |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderCRUDController.java b/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderCRUDController.java
index 9d07072..dc2be48 100644
--- a/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderCRUDController.java
+++ b/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderCRUDController.java
@@ -106,8 +106,13 @@ public class OrderCRUDController extends GenericForwardComposer {
                                 if (invalidValue.getBean() instanceof OrderElement) {
                                     Label result = new Label();
 
-                                    String orderElementName = ((OrderElement) invalidValue
-                                            .getBean()).getName();
+                                    String orderElementName;
+                                    if (invalidValue.getBean() instanceof Order) {
+                                        orderElementName = _("Order");
+                                    } else {
+                                        orderElementName = ((OrderElement) invalidValue
+                                                .getBean()).getName();
+                                    }
 
                                     result.setValue(orderElementName + " "
                                             + invalidValue.getPropertyName()
-- 
1.6.0.4

