From 99cc3030e5c253d4594a2341b75d5b730906baf5 Mon Sep 17 00:00:00 2001
From: Manuel Rego Casasnovas <mrego@igalia.com>
Date: Wed, 14 Oct 2009 10:49:27 +0200
Subject: [PATCH] ItEr30S17ValidacionEProbasFuncionais: Fixing problem indenting order elements with arrows. Fixes Bug #7.

---
 .../web/orders/OrderElementTreeController.java     |   42 +++++++++++++-------
 1 files changed, 28 insertions(+), 14 deletions(-)

diff --git a/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderElementTreeController.java b/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderElementTreeController.java
index b86d701..91afb97 100644
--- a/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderElementTreeController.java
+++ b/navalplanner-webapp/src/main/java/org/navalplanner/web/orders/OrderElementTreeController.java
@@ -92,12 +92,16 @@ public class OrderElementTreeController extends GenericForwardComposer {
     }
 
     public void indent() {
-        snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
         if (tree.getSelectedCount() == 1) {
-            getModel().indent(getSelectedNode());
+            indent(getSelectedNode());
         }
     }
 
+    private void indent(OrderElement orderElement) {
+        snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
+        getModel().indent(orderElement);
+    }
+
     public TreeModel getOrderElementTreeModel() {
         if (getModel() == null) {
             return null;
@@ -110,26 +114,39 @@ public class OrderElementTreeController extends GenericForwardComposer {
     }
 
     public void unindent() {
-        snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
         if (tree.getSelectedCount() == 1) {
-            getModel().unindent(getSelectedNode());
+            unindent(getSelectedNode());
         }
     }
 
+    private void unindent(OrderElement orderElement) {
+        snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
+        getModel().unindent(orderElement);
+    }
+
     public void up() {
         snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
         if (tree.getSelectedCount() == 1) {
-            getModel().up(getSelectedNode());
+            up(getSelectedNode());
         }
     }
 
-    public void down() {
+    public void up(OrderElement orderElement) {
         snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
+        getModel().up(orderElement);
+    }
+
+    public void down() {
         if (tree.getSelectedCount() == 1) {
-            getModel().down(getSelectedNode());
+            down(getSelectedNode());
         }
     }
 
+    public void down(OrderElement orderElement) {
+        snapshotOfOpenedNodes = TreeViewStateSnapshot.snapshotOpened(tree);
+        getModel().down(orderElement);
+    }
+
     private OrderElement getSelectedNode() {
         return (OrderElement) tree.getSelectedItemApi().getValue();
     }
@@ -445,7 +462,7 @@ public class OrderElementTreeController extends GenericForwardComposer {
             upbutton.addEventListener(Events.ON_CLICK, new EventListener() {
                 @Override
                 public void onEvent(Event event) throws Exception {
-                    getModel().down(orderElementForThisRow);
+                    down(orderElementForThisRow);
                 }
             });
 
@@ -457,7 +474,7 @@ public class OrderElementTreeController extends GenericForwardComposer {
             downbutton.addEventListener(Events.ON_CLICK, new EventListener() {
                 @Override
                 public void onEvent(Event event) throws Exception {
-                    getModel().up(orderElementForThisRow);
+                    up(orderElementForThisRow);
                 }
             });
 
@@ -469,8 +486,7 @@ public class OrderElementTreeController extends GenericForwardComposer {
             indentbutton.addEventListener(Events.ON_CLICK, new EventListener() {
                 @Override
                 public void onEvent(Event event) throws Exception {
-                    getModel().indent(orderElementForThisRow);
-
+                    indent(orderElementForThisRow);
                 }
             });
 
@@ -483,9 +499,7 @@ public class OrderElementTreeController extends GenericForwardComposer {
                     new EventListener() {
                         @Override
                         public void onEvent(Event event) throws Exception {
-                            getModel().unindent(
-                                    orderElementForThisRow);
-
+                            unindent(orderElementForThisRow);
                         }
                     });
 
-- 
1.6.0.4

